### DISCUSSION TITLE
API Endpoint Testing

### DISCUSSION THREAD

**Discussion by cdnicoll** [noise-lab/iot-inspector-client#126] [Q&A]
## API Endpoints
- ✅ http://127.0.0.1:53721/api/get_device_list
  - Working with mock data, returns a device list
  - Might need a way to see if the device should be monitored or not, as per this [screen](https://xd.adobe.com/view/46937def-3c9f-4909-8a76-308baac2bcf0-13c0/screen/ded3f6e5-d8ff-498e-8ec7-134b348bcbc3/)
- ✅ http://127.0.0.1:53721/api/get_overall_device_stats/DeviceA/86400
  - Working with mock data
- ❌ http://127.0.0.1:53721/api/get_device_network_activities/DeviceA
  - Not returning what we expect
  - Might need mock data
- 🟠 http://127.0.0.1:53721/api/get_all_counterparties/86400
  - Returning an empty result set
  - Will need mock data
- ✅ http://127.0.0.1:53721/api/get_global_config
  - Returns mock data
- ❌ http://127.0.0.1:53721/api/set_global_config
  - Endpoint exists
  - Not sure if it's been completed yet?
  - Doesn't look like it takes any key/value options
- ❌ http://127.0.0.1:53721/api/set_custom_option/{option_key}/{option_value}
  - Need a list of option keys and their values
- ✅ http://127.0.0.1:53721/api/get_overall_bandwidth_consumption
  - Data is in place
- ✅ http://127.0.0.1:53721/api/download_all_data 
  - Basic Mock data is in place

## What else might be needed
- We're pulling data on the dashboard pretty regularly as we monitor and display device information. Ideally this will need to be as up to date as possible.
  - we could hit the API every several seconds
  - use of websockets?
  - graphql subscriptions
- We have the idea of an [achievement screen](https://xd.adobe.com/view/250bfa62-c34a-4356-9cf3-dedf1313c814-7058/screen/b3ca1ddc-803f-4421-81ee-beccc6e88813/)
  - It's currently proposed we calculate these client side, if the user resets or wipes their information all achievements may be lost. Do we want to store this as an API endpoint?
- Monitoring devices, we have a checkbox to track or not, can we use the `/set_device_state` endpoint for this?

**Comment by crazyideas21** [comment]
I can switch to graphql (rather than REST API) to reduce the number of requests and the sizes of responses, as the UI can fetch exactly what it wants, a la carte.
